src/main/scala/com/microsoft/hyperspace/index/rules/FilterIndexRule.scala

```diff
@@ -56,6 +57,11 @@ object FilterIndexRule
             findCoveringIndexes(filter, outputColumns, filterColumns)
           FilterIndexRanker.rank(spark, filter, candidateIndexes) match {
             case Some(index) =>
+              val useBucketSpec = if (HyperspaceConf.filterRuleBucketCheckEnabled(spark)) {
+                BucketSelector(plan, index.bucketSpec).isDefined
+              } else {
+                HyperspaceConf.useBucketSpecForFilterRule(spark)
```

Comment:
This can be a confusing experience since `useBucketSpec` even when `spark.hyperspace.index.filterRule.useBucketSpec` is set to true.

How about:
```scala
if (HyperspaceConf.useBucketSpecForFilterRule(spark)) {
   // always use bucket spec
} else {
  val selectedBuckets = BucketSelector(plan, index.bucketSpec)
  selectedBuckets.isDefined && (getSelectedRatio(selectedBuckets) < HyperspaceConf. prunedBucketRatioToAutoEnableBucketRead)
}
```